ui/components/NodeInputSubmit.tsx

```diff
@@ -33,39 +33,55 @@ export const NodeInputSubmit: FC<NodeInputProps> = ({
   const isProvider = attributes.name === "provider";
   const provider = attributes.value as string;
   const image = getProviderImage(provider);
+  const showBackupLink =
+    (node.meta.label as unknown as { hasBackupLink: boolean })?.hasBackupLink ??
+    false;
 
   return (
-    <Button
-      appearance={
-        node.group === "password" ||
-        node.group === "code" ||
-        node.group === "totp" ||
-        node.group === "webauthn" ||
-        node.group === "lookup_secret"
-          ? "positive"
-          : ""
-      }
-      onClick={async (e) => {
-        // On click, we set this value, and once set, dispatch the submission!
-        await setValue(attributes.value as string).then(() =>
-          dispatchSubmit(e),
-        );
-      }}
-      disabled={attributes.disabled || disabled}
-      className={node.group === "oidc" ? "oidc-login-button" : ""}
-    >
-      {isProvider ? (
-        <>
-          <img
-            src={image}
-            alt={`${provider} logo`}
-            style={{ marginRight: "0.5rem" }}
-          />
-          <span>Sign in with {label}</span>
-        </>
-      ) : (
-        label
+    <>
+      <Button
+        appearance={
+          node.group === "password" ||
+          node.group === "code" ||
+          node.group === "totp" ||
+          node.group === "webauthn" ||
+          node.group === "lookup_secret"
+            ? "positive"
+            : ""
+        }
+        onClick={async (e) => {
+          // On click, we set this value, and once set, dispatch the submission!
+          await setValue(attributes.value as string).then(() =>
+            dispatchSubmit(e),
+          );
+        }}
+        disabled={attributes.disabled || disabled}
+        className={node.group === "oidc" ? "oidc-login-button" : ""}
+      >
+        {isProvider ? (
+          <>
+            <img
+              src={image}
+              alt={`${provider} logo`}
+              style={{ marginRight: "0.5rem" }}
+            />
+            <span>Sign in with {label}</span>
+          </>
+        ) : (
+          label
+        )}
+      </Button>
+      {showBackupLink && (
+        <Button
+          appearance="link"
+          type="button"
+          onClick={() =>
+            (window.location.href = window.location.href + "&use_backup_code=1")
```

Comment:
question: I would expect a `true` value instead of `1`, just curious to know why we have an int here

Reply:
changing it to true.